cypress/e2e/ui/Overview/Chargeback/rates.cy.js

```diff
@@ -0,0 +1,890 @@
+      cy.expect_text(':nth-child(9) > :nth-child(2) > .cell > .array_list > :nth-child(2)', 'Memory Used');
+      cy.expect_text(':nth-child(9) > :nth-child(3) > .cell > .array_list > .list_row', '0.0');
+      cy.expect_text(':nth-child(9) > :nth-child(4) > .cell > .array_list > .list_row', 'Infinity');
+      cy.expect_text(':nth-child(9) > :nth-child(5) > .cell > .array_list > .list_row', '0.0');
+      cy.expect_text(':nth-child(9) > :nth-child(6) > .cell > .array_list > .list_row', '0.02');
+      cy.get(':nth-child(9) > :nth-child(7) > .cell > .bx--front-line').contains(currency);
+      cy.get(':nth-child(9) > :nth-child(7) > .cell > .bx--front-line').contains('Hour / MB');
+
+      cy.expect_text(':nth-child(10) > :nth-child(1) > .cell > .bx--front-line', 'Network I/O');
+      cy.expect_text(':nth-child(10) > :nth-child(2) > .cell > .array_list > :nth-child(1)', 'Used Network I/O');
+      cy.expect_text(':nth-child(10) > :nth-child(2) > .cell > .array_list > :nth-child(2)', 'Network I/O Used');
+      cy.expect_text(':nth-child(10) > :nth-child(3) > .cell > .array_list > .list_row', '0.0');
+      cy.expect_text(':nth-child(10) > :nth-child(4) > .cell > .array_list > .list_row', 'Infinity');
+      cy.expect_text(':nth-child(10) > :nth-child(5) > .cell > .array_list > .list_row', '0.5');
+      cy.expect_text(':nth-child(10) > :nth-child(6) > .cell > .array_list > .list_row', '0.0');
+      cy.get(':nth-child(10) > :nth-child(7) > .cell > .bx--front-line').contains(currency);
+      cy.get(':nth-child(10) > :nth-child(7) > .cell > .bx--front-line').contains('Hour / KBps');
+    }).then(() => {
+      cy.get('#chargeback_rate_vmdb_choice').click();
+      cy.get(':nth-child(1) > .bx--overflow-menu-options__btn').click();
+      cy.get('#description').type('{moveToEnd} edit');
+      cy.get('.btn-group > .btn').click();
+      cy.get('[data-original-index="144"]').then((option) => {
+        currency = option[0].innerText;
+        cy.get(option).click();
+      });
+      cy.get('#per_time_0').select('Daily');
+      cy.get('#finish_0_0').type('100');
+      cy.get('#fixed_rate_0_0').clear().type('2.0');
+      cy.get('#rate_detail_row_0_0 > .action-cell > .btn').click();
+      cy.get('#fixed_rate_0_1').clear().type('10.0');
+      cy.get('#variable_rate_0_0').clear().type('10');
+      cy.get('#variable_rate_0_1').clear().type('100');
+
+      cy.get('#rate_detail_row_1_1 > .action-cell > .btn').click();
+
+      cy.get('#per_time_4').select('Hourly');
+      cy.get('#per_unit_4').select('Bps');
+
+      cy.get('#finish_1_0').clear();
+
+      cy.get('#buttons_on > .btn-primary').click({ force: true });
+    }).then(() => {
+      cy.get(':nth-child(1) > .col-md-8 > .form-control-static').contains('Cypress test compute chargeback rates edit');
+      cy.get(':nth-child(2) > .col-md-8 > .form-control-static').contains('Compute');
+
+      cy.get('thead > tr > :nth-child(1)').contains('Group');
```

Comment:
It feels like we could move this pattern into a "expect_table" custom method that knows how to walk the table structure.  Then I could see something like here being:

```js
cy.expect_table(
  header: ['Group', 'Description (Column Name in Report)', ...],
  rows: [
    [0.0, 100.0, 100.0 ...],
    [...]
  ]
```

Reply:
Yeah that's a good idea, I can push that up in a new pr